resources/assets/js/components/teams/TeamMembershipButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <div>
+        <button
+                type="button" class="btn btn-secondary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-if="memberOfTeam" v-on:click="changeMembership('leave')">
+            Leave
+        </button>
+        <button
```

Comment:
Do not display a join/leave button if the user cannot join/leave the team.

Note that this will make the grey area at the bottom of the box a different size than the ones with buttons unless you do something to fix that. I think that's fine.